### PR TITLE
Added IFV passenger dependent turrets

### PR DESCRIPTION
**Pull request by Mailaender**
A prime example of https://github.com/OpenRA/OpenRA/wiki/Actor-Upgrades

Depends on https://github.com/OpenRA/OpenRA/pull/10567 for the Travis check to pass.

### CODE REVIEW

**Comment by GraionDilach** [comment]
I'm not a fan of this method but indeed, the Tick Tank PR didn't got merged (which would have allowed `AttackTurreted` be upgraded instead of `Armament`s to allow eliteweapons on the additional IFV modes).

File: sequences/voxels.yaml [hunk 1]
```diff
@@ -65,9 +65,21 @@ naflak:
...
+	turret-advm: fvtur3
+	turret-sniper: fvtur4
+	turret-chrono: fvtur5
+	turret-flak: fvtur6
+	turret-tesla: fvtur7
+	turret-demo: fvtur8
+	turret-deso: fvtur9
+	turret-psi: fvtur10
+	turret-hero: fvtur11
+	turret-engi: fvtur12
```
Comment (MustaphaTR) [inline sequences/voxels.yaml]: These are all wrong there are only 4 turret that really used, Missile, MG, Repair and Hi-Tech. Which are `fvtur`, `fvtur1`, `fvtur2` and `fvtur3`. Others are unused.

Comment (GraionDilach) [inline sequences/voxels.yaml]: The Cow Turret was used. IIRC the rest are duplicate files so no harm done.

Comment (MustaphaTR) [inline sequences/voxels.yaml]: Having them here doesn't harm but they are also defined like that under `WithVoxelTurret:`.

Comment (Mailaender) [inline sequences/voxels.yaml]: That fixes the glitches. Thanks guys.

**Comment by abcdefg30** [comment]
OpenRA/OpenRA#10567 was merged.

**Comment by Mailaender** [comment]
Rebased.

**Comment by Iamgoofball** [comment]
Doesn't the IFV also get improved move speed if you stuff a dog in there?

**Comment by GraionDilach** [comment]
RA2 never had such a logic at all.

**Comment by abcdefg30** [comment]
Looks good to me. In the original other units (like the spy) gave the IFV a `machinegun` turret, too. But I'm not certain anymore and will recheck that. In addition this pr is already good enough, so :+1: / :white_check_mark:

**Comment by Iamgoofball** [comment]
Mighta been YR then, idk, I swear dogs in IFVs would make them go faster

**Comment by GraionDilach** [comment]
Don't swear. Such logic was impossible to do in the engine. Passengers couldn't affect stats of their cargo at all. The IFV was one giant hackish approach in WW and no, it only could change turret and weapon, never speed or any other stat, neither in RA2 neither in YR.

**Pull request merged**